code/game/objects/items/RCD.dm

```diff
@@ -554,27 +555,53 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
 	 *RCD_WALLFRAME is also returned as the mode when upgrading apc, airalarm, firealarm using simple circuits upgrade
 	 */
 	if(rcd_results["mode"] != RCD_WALLFRAME && rcd_results["mode"] != RCD_DECONSTRUCT)
-		//if we are trying to build a window on top of a grill [only if there isnt already a window there] then we skip this check because thats normal behaviour
-		if(rcd_results["mode"] == RCD_WINDOWGRILLE && locate(/obj/structure/grille, target_turf) && (!locate(/obj/structure/window, target_turf) && !locate(/obj/structure/window/reinforced, target_turf)))
+		//if we are trying to build a window on top of a grill then we skip this check because thats normal behaviour
+		if(rcd_results["mode"] == RCD_WINDOWGRILLE && istype(A, /obj/structure/grille))
+			//no checks proceed
```

Comment:
```suggestion
		if(rcd_results["mode"] == RCD_WINDOWGRILLE && istype(A, /obj/structure/grille)) {}
			//no checks proceed
```
byond doesn't like empty if blocks, you can work around it by manually declaring an empty block though

Reply:
Actually nevermind, your run failures are from rust-g dying horribly

Reply:
Its fine we will keep the blocks